### PR TITLE
Support number of centroids in approx_percentile_cont

### LINKED ISSUE #3142: Support create ApproxPercentileAccumulator with TDigest max_size
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
We are using ApproxPercentileAccumulator to create self define udaf to calculate ApproxPercentile from pre-calculated data.
 
But i found in df
```rust
impl ApproxPercentileAccumulator {
    pub fn new(percentile: f64, return_type: DataType) -> Self {
        Self {
            digest: TDigest::new(DEFAULT_MAX_SIZE),
            percentile,
            return_type,
        }
    }
```
the DEFAULT_MAX_SIZE is always 100,
according to the `t-digest` 
> Number of centroids(sample data) is greater than `maxSize`, we need to compress them
 When merging

means when facing large amount data using DEFAULT_MAX_SIZE is not suitble.
like in hive the default is 10,000
https://github.com/apache/hive/blob/7b3ecf617a6d46f48a3b6f77e0339fd4ad95a420/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFPercentileApprox.java#L59-L62
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #3145: Support number of histogram bins in approx_percentile_cont
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
like apache-hive 
```
@Description(name = "percentile_approx",
    value = "_FUNC_(expr, pc, [nb]) - For very large data, computes an approximate percentile " +
            "value from a histogram, using the optional argument [nb] as the number of histogram" +
            " bins to use. A higher value of nb results in a more accurate approximation, at " +
            "the cost of higher memory usage.",
    extended = "'expr' can be any numeric column, including doubles and floats, and 'pc' is " +
               "either a single double/float with a requested percentile, or an array of double/" +
               "float with multiple percentiles. If 'nb' is not specified, the default " +
               "approximation is done with 10,000 histogram bins, which means that if there are " +
               "10,000 or fewer unique values in 'expr', you can expect an exact result. The " +
               "percentile() function always computes an exact percentile and can run out of " +
               "memory if there are too many unique values in a column, which necessitates " +
               "this function.\n" +
               "Example (three percentiles requested using a finer histogram approximation):\n" +
               "> SELECT percentile_approx(val, array(0.5, 0.95, 0.98), 100000) FROM somedata;\n" +
               "[0.05,1.64,2.26]\n")
```
Need add args max_size of tdigest(aka histogram bins) in approx_percentile_cont,
when calculate large amount data we need more sample points to keep accuracy.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Closes #3142
Closes #3145 

Now we support approx_percentile_cont(col_name, percentile value, number of centroids)

```
DataFusion CLI v10.0.0
❯ create external table order stored as parquet location '/Users/yangjiang/test-data/tpch-1g/orders';
0 rows in set. Query took 0.016 seconds.
❯ select o_orderstatus, approx_percentile_cont(o_totalprice, 0.5) from order group by o_orderstatus  order by 1 limit 3;
+---------------+-------------------------------------------------------+
| o_orderstatus | APPROXPERCENTILECONT(order.o_totalprice,Float64(0.5)) |
+---------------+-------------------------------------------------------+
| F             | 143415.09074751436                                    |
| O             | 143177.55673372833                                    |
| P             | 181407.62680871075                                    |
+---------------+-------------------------------------------------------+
3 rows in set. Query took 0.728 seconds.
❯ select o_orderstatus, approx_percentile_cont(o_totalprice, 0.5, 10000) from order group by o_orderstatus  order by 1 limit 3;
+---------------+--------------------------------------------------------------------+
| o_orderstatus | APPROXPERCENTILECONT(order.o_totalprice,Float64(0.5),Int64(10000)) |
+---------------+--------------------------------------------------------------------+
| F             | 143289.22569288642                                                 |
| O             | 143184.3388358634                                                  |
| P             | 181451.707578125                                                   |
+---------------+--------------------------------------------------------------------+
3 rows in set. Query took 1.168 seconds.
❯ 


```

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs [hunk 1]
```diff
@@ -113,6 +116,64 @@ impl ApproxPercentileCont {
...
+        .as_any()
+        .downcast_ref::<Literal>()
+        .ok_or_else(|| {
+            DataFusionError::Internal(
+                "desired percentile argument must be float literal".to_string(),
+            )
+        })?
+        .value();
+    let max_size = match lit {
+        ScalarValue::UInt8(Some(q)) => *q as usize,
```
Comment (Ted-Jiang) [inline datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs:161]: For now i think ew can not input unsigned int from sql 🤔 
So need these Uint.

**Comment by Ted-Jiang** [comment]
@alamb PTAL😊

**Comment by andygrove** [comment]
We should update the documentation at https://github.com/apache/arrow-datafusion/blob/master/docs/source/user-guide/sql/aggregate_functions.md as part of this PR

**Review by alamb — COMMENTED** [review]
Thank you @Ted-Jiang  -- @domodwyer  / @realno  I wonder if you have any interest in reviewing this PR.

**Comment by Ted-Jiang** [comment]
> We should update the documentation at https://github.com/apache/arrow-datafusion/blob/master/docs/source/user-guide/sql/aggregate_functions.md as part of this PR

Thanks your remind @andygrove  add in [7d67a2f](https://github.com/apache/arrow-datafusion/pull/3146/commits/7d67a2fd4f4d266d78891f93eebdd4ec2db8fbd1)

File: datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs [hunk 2]
```diff
@@ -52,39 +53,35 @@ impl ApproxPercentileCont {
...
-            .value();
-        let percentile = match lit {
-            ScalarValue::Float32(Some(q)) => *q as f64,
-            ScalarValue::Float64(Some(q)) => *q as f64,
-            got => return Err(DataFusionError::NotImplemented(format!(
-                "Percentile value for 'APPROX_PERCENTILE_CONT' must be Float32 or Float64 literal (got data type {})",
-                got
-            )))
-        };
+        let percentile = validate_input_percentile_expr(&expr[1])?;
```
Comment (alamb) [inline datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs:56]: 👍

File: datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs (hunk 1 above)
Comment (alamb) [reply datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs:161]: Yeah, I think we should file a ticket to allow creating unsigned int types from SQL (like in `CREATE TABLE`) statements.

I ran into this limitation while I was fixing https://github.com/apache/arrow-datafusion/pull/3167 -- https://github.com/apache/arrow-datafusion/pull/3167/files#r946145318

File: datafusion/core/tests/sql/aggregates.rs [hunk 3]
```diff
@@ -925,6 +925,54 @@ async fn csv_query_approx_percentile_cont_with_weight() -> Result<()> {
...
+#[tokio::test]
+async fn csv_query_approx_percentile_cont_with_histogram_bins() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_aggregate_csv(&ctx).await?;
+
+    // compare approx_percentile_cont and approx_percentile_cont_with_weight
+    let sql = "SELECT c1, approx_percentile_cont(c3, 0.95, 200) AS c3_p95 FROM aggregate_test_100 GROUP BY 1 ORDER BY 1";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----+--------+",
```
Comment (alamb) [inline datafusion/core/tests/sql/aggregates.rs:937]: I am not a statistics expert but I verified that this is different than the output generated by `SELECT c1, approx_percentile_cont_with_weight(c3, c2, 0.95) AS c3_p95 FROM aggregate_test_100 GROUP BY 1 ORDER BY 1` a few lines above

**Review by alamb — APPROVED** [review]
Nice work @Ted-Jiang  -- I think the code and tests look quite nice.

Sorry for the delay in review

File: datafusion/core/tests/sql/aggregates.rs (hunk 3 above)
Comment (Ted-Jiang) [reply datafusion/core/tests/sql/aggregates.rs:937]: @alamb Thanks for your review 😊
this sql should have with the same result as 
`SELECT c1, approx_percentile_cont(c3, 0.95) AS c3_p95 FROM aggregate_test_100 GROUP BY 1 ORDER BY 1`
in line 867

**Pull request merged**